misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt

```diff
@@ -18,12 +19,12 @@ private val logger = getLogger<TracingInterceptor>()
 internal class TracingInterceptor internal constructor(private val tracer: Tracer): Interceptor {
   @Singleton
   class Factory : Interceptor.Factory {
-    @Inject(optional=true) var tracer: Tracer? = null
+    @Inject lateinit var tracer: Tracer
 
     override fun create(action: Action): Interceptor? {
       // NOTE(nb): returning null ensures interceptor is filtered out when generating interceptors to
       // apply for a specific action. See WebActionModule for implementation details
-      return if (tracer != null) TracingInterceptor(tracer!!) else null
+      return if (tracer is NoopTracer) null else TracingInterceptor(tracer)
```

Comment:
why the special handling? shouldn't you be able to dispatch to the `NoopTracer` and it do nothing?